### PR TITLE
npm_registry key in yaml that is then written to NPM_CONFIG_REGISTRY env var

### PR DESCRIPTION
**Pull request by igorwwwwwwwwwwwwwwwwwwww**

### CODE REVIEW

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Ping @BanzaiMan, could you take a look? Do any other places need to be changed for this to work? [travis-yaml](https://github.com/travis-ci/travis-yaml)? Also, would you like to have tests for this patch -- and if so can you give me a hint for how to go about it?

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
oh, and is there a good way to do echo escaping?

**Comment by BanzaiMan** [comment]
Where do we mention `npm_registry` key? I'd never heard of this option.

**Comment by BanzaiMan** [comment]
In general, appliances are for things that we configure and enforce. Addons are for user-defined configurations.

**Comment by joshk** [comment]
I think the use of appliance is correct then.

**Comment by BanzaiMan** [comment]
@joshk I don't think so. This will make whatever users put in `npm_registry` key value available to PRs, even when it is encrypted.

**Comment by BanzaiMan** [comment]
Oh, that's not true. Disregard that.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
To give you some more context, the behaviour of this will likely change over time. I wanted to start with a yml key, then move to user/org, then % of traffic, then globally.

And also, once we have a more solid setup, it will become a boolean flag as opposed to passing the actual repository URL -- the repository URL will be hard-coded or set as an env var.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Updated with comments as per @joshk. Once this is merged we can put a feature toggle into scheduler along the lines of `cache_settings`: https://github.com/travis-ci/travis-scheduler/blob/master/lib/travis/scheduler/payloads/worker.rb#L46

File: lib/travis/build/appliances/npm_registry.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'travis/build/appliances/base'
+
+module Travis
+  module Build
+    module Appliances
+      class NpmRegistry < Base
+        def apply
+          if data[:npm_registry]
```
Comment (BanzaiMan) [inline lib/travis/build/appliances/npm_registry.rb]: This logic should be moved to `apply?`.

**Comment by BanzaiMan** [comment]
Needs specs.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
@BanzaiMan updated with `apply?` and specs!

**Pull request merged**